sentence_transformers/util.py

```diff
@@ -655,18 +658,27 @@ def mine_hard_negatives(
 
     log_counters = {}
     queries = dataset[columns[0]]
-    corpus = dataset[columns[1]]
+    positives = dataset[columns[1]]
+    separate_corpus = corpus is not None
+    if not separate_corpus:
+        corpus = positives
 
-    # Embed the corpus and queries
+    # Embed the corpus, queries, and positives
     corpus_embeddings = model.encode(corpus, batch_size=batch_size, convert_to_tensor=True, show_progress_bar=True)
     query_embeddings = model.encode(queries, batch_size=batch_size, convert_to_tensor=True, show_progress_bar=True)
+    if separate_corpus:
+        positives_embeddings = model.encode(
+            positives, batch_size=batch_size, convert_to_tensor=True, show_progress_bar=True
+        )
+    else:
+        positives_embeddings = corpus_embeddings
```

Comment:
If we don't have a separate corpus, then we don't have to reembed the positives :)